src/ripple/app/main/Application.cpp

```diff
@@ -1227,11 +1227,18 @@ bool ApplicationImp::setup()
     }
 
     {
-        auto setup = setup_ServerHandler(
-            *config_,
-            beast::logstream { m_journal.error() });
-        setup.makeContexts();
-        serverHandler_->setup (setup, m_journal);
+        try
+        {
+            auto setup = setup_ServerHandler(
+                *config_, beast::logstream{m_journal.error()});
+            setup.makeContexts();
+            serverHandler_->setup(setup, m_journal);
+        }
+        catch (std::exception const&)
+        {
+            JLOG(m_journal.fatal()) << "Unable to setup server handler";
```

Comment:
It might be useful to log the exception message if one exists. Even if the current code is unlikely to produce one now, it might catch a new message in the future.

Reply:
Good suggestions. I went with printing the message if one exists to keep a limited scope of this PR.